addons/project/models/project_project.py

```diff
@@ -440,6 +432,11 @@ def create(self, vals_list):
             if stage.company_id:
                 for vals in vals_list:
                     vals['company_id'] = stage.company_id.id
+        for vals in vals_list:
+            if 'is_favorite' in vals:
+                if vals['is_favorite']:
+                    vals['favorite_user_ids'] = [self.env.uid]
+                vals.pop('is_favorite')
```

Comment:
You can combine both checks into one with
```suggestion
            if vals.pop('is_favorite', False):
                vals['favorite_user_ids'] = [self.env.uid]
```
You could argue that it's a bit less readable though, so up to you